viewing_party/party.py

```diff
@@ -0,0 +1,214 @@
+            unique_watch_list.append(user_movie)
+        else:
+            pass
+    
+    return unique_watch_list
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = user_data["watched"]
+    
+    friends_list = user_data["friends"]
+    friends_watched_lists = []
+    list_of_friends_movies = []
+    no_dups = []
+
+    user_not_watched = []
+
+# Isolate and extract lists/dicts
+
+    for friend in friends_list:
+        for watched_lists in friend.values():
+            friends_watched_lists.append(watched_lists)
+    
+    for list in friends_watched_lists:
+        for movie in list:
+            list_of_friends_movies.append(movie)
+
+    for movie in list_of_friends_movies:
+        if movie not in no_dups:
+            no_dups.append(movie)
+        else:
+            pass
+    
+    for friend_movie in no_dups:
+        if friend_movie not in user_watched_list:
+            user_not_watched.append(friend_movie)
+        else:
+            pass
+    
+    return user_not_watched
+
+def get_available_recs(user_data):
+    user_watched_list = user_data["watched"]
+    friends_list = user_data["friends"]
+    friends_watched_lists = []
+    list_of_friends_movies = []
+    no_dups = []
+    watched = None
+    user_not_watched = []
+    available_recs = []
+
+# Isolate and extract lists/dicts
+
+    for friend in friends_list:
+        for watched_lists in friend.values():
+            friends_watched_lists.append(watched_lists)
+    
+    for list in friends_watched_lists:
+        for movie in list:
+            list_of_friends_movies.append(movie)
+
+    for movie in list_of_friends_movies:
+        if movie not in no_dups:
+            no_dups.append(movie)
+        else:
+            pass
+    
+    for movie in no_dups:
+        watched = False
+        for user_movie in user_watched_list:
+            if user_movie["title"] == movie["title"]:
+                watched = True
+        if watched == False:
+            user_not_watched.append(movie)
+    
+    for movie in user_not_watched:
+        if movie["host"] in user_data["subscriptions"]:
+            available_recs.append(movie)
+    
+    return available_recs
+
+def get_new_rec_by_genre(user_data):
+    freq_genre = get_most_watched_genre(user_data)
+    user_not_watched = get_friends_unique_watched(user_data)
+    rec_list = []
+
+    for movie in user_not_watched:
+        if movie["genre"] == freq_genre:
+            rec_list.append(movie)
+    
+    return rec_list
+
+def get_rec_from_favorites(user_data):
+    user_favs = user_data["favorites"]
+    friends_list = user_data["friends"]
+    list_of_friends_movies = []
+    fav_recs = []
+
+    for friend in friends_list:
+        for movie in friend["watched"]:
+            list_of_friends_movies.append(movie)
+    
+    for movie in user_favs:
+        if movie not in list_of_friends_movies:
+            fav_recs.append(movie)
+    
+    return fav_recs
```

Comment:
💡 Quick-Tip: To make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.